Python/sysmodule.c

```diff
@@ -1021,6 +1021,43 @@ Set the global debug tracing function.  It will be called on each\n\
 function call.  See the debugger chapter in the library manual."
 );
 
+/*[clinic input]
+sys._settraceallthreads
+
+    arg: object
+    /
+
+Set the global debug tracing function in all running threads.
+
+It will be called on each function call. See the debugger chapter
+in the library manual.
+[clinic start generated code]*/
+
+static PyObject *
+sys__settraceallthreads(PyObject *module, PyObject *arg)
+/*[clinic end generated code: output=161cca30207bf3ca input=57c1ae02b1b5ee94]*/
+{
+    PyObject* argument = NULL;
+    Py_tracefunc func = NULL;
+
+    if (arg != Py_None) {
+        func = trace_trampoline;
+        argument = arg;
+    }
+
+    PyThreadState *this_tstate = _PyThreadState_GET();
+    PyInterpreterState* interp = this_tstate->interp;
+    PyThreadState* ts = PyInterpreterState_ThreadHead(interp);
+    while (ts) {
+        if (_PyEval_SetTrace(ts, func, argument) < 0) {
+            return NULL;
+        }
+        ts = PyThreadState_Next(ts);
```

Comment:
I would prefer to use HEAD_LOCK(runtime) in all functions or no functions.

Reply:
I'm going to use the C-APIs directly here for simplicity